app/lib/bulk_download_service.rb

```diff
@@ -184,4 +208,52 @@ def self.generate_output_path_map(study_files)
     end
     output_map
   end
+
+  # generate a study_info.json object from an existing study
+  def self.generate_study_manifest(study, hostname)
+    info = HashWithIndifferentAccess.new
+    info[:study] = {
+      name: study.name,
+      description: study.description.try(:truncate, 150),
+      accession: study.accession,
+      cell_count: study.cell_count,
+      gene_count: study.gene_count,
+      link: hostname + Rails.application.routes.url_helpers.view_study_path(accession: study.accession, study_name: study.name)
+    }
+    info[:files] = study.study_files
+                        .where(queued_for_deletion: false)
+                        .map{|f| generate_study_file_manifest(f)}
+    info
+  end
+
+  # generate a study_info.json object from an existing study_file
+  def self.generate_study_file_manifest(study_file)
+    output = {
+      filename: study_file.name,
+      file_type: study_file.file_type
+    }
+
+    if study_file.expression_file_info
+      # explicitly list properties to copy to avoid accidentally exposing internal fields
+      props_to_copy =  [
```

Comment:
I don't understand why you need to list this out if you're passing them all directly to the output.  This is only going to create a place that will get out of sync with the model (especially since I'm renaming `library_construction_protocol` to `library_preparation_protocol` in my branch already).  

You can achieve the same thing with this:
```
study_file.expression_file_info.attributes.each {|key, value| output[key] = value}
```